src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/segmentgroup/LabResult.java

```diff
@@ -0,0 +1,89 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.List;
+import java.util.Optional;
+
+import lombok.Getter;
+import lombok.NonNull;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.DiagnosticReportCode;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.LaboratoryInvestigation;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.LaboratoryInvestigationResult;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.MessageTrailer;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.SequenceDetails;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.SequenceReference;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.ServiceProviderCommentFreeText;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.TestStatus;
+
+/**
+ * Provides information about laboratory investigation results.
+ * <p>
+ * Segment group 18: {@code GIS-INV-SEQ-RSL-STS-DTM-FTX-RFF-SG19-SG20}
+ * <p>
+ * Parents: {@link uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message Message}
+ * &gt; {@link ServiceReportDetails}
+ * &gt; {@link InvestigationSubject}
+ */
+@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+public class LabResult extends SegmentGroup {
+    public static final String INDICATOR = DiagnosticReportCode.KEY;
+
+    // GIS
+    @Getter(lazy = true)
+    private final DiagnosticReportCode diagnosticReportCode =
+        DiagnosticReportCode.fromString(extractSegment(DiagnosticReportCode.KEY));
+
+    // INV+MQ
+    @Getter(lazy = true)
+    private final LaboratoryInvestigation laboratoryInvestigation =
+        LaboratoryInvestigation.fromString(extractSegment(LaboratoryInvestigation.KEY_QUALIFIER));
+
+    // SEQ?
+    @Getter(lazy = true)
+    private final Optional<SequenceDetails> sequenceDetails =
+        extractOptionalSegment(SequenceDetails.KEY)
+            .map(SequenceDetails::fromString);
+
+    // RSL+NV?
+    @Getter(lazy = true)
+    private final Optional<LaboratoryInvestigationResult> laboratoryInvestigationResult =
+        extractOptionalSegment(LaboratoryInvestigationResult.KEY_QUALIFIER)
+            .map(LaboratoryInvestigationResult::fromString);
+
+    // STS?
+    @Getter(lazy = true)
+    private final Optional<TestStatus> testStatus =
+        extractOptionalSegment(TestStatus.KEY)
+            .map(TestStatus::fromString);
+
+    // FTX{,99}
+    @Getter(lazy = true)
+    private final List<ServiceProviderCommentFreeText> serviceProviderCommentFreeTexts =
```

Comment:
> Is there a way to make the use of an optional dependent upon certain conditions?
>
> The word document spec states that this section is dependent on certain rules being met for it to be used so I'm worried about it not being optional.

There are validation libraries that support that kind of thing, but we aren't currently importing them.

> Also I think we need to refactor this segment so that `SPC`, `RIT` and `CRR` qualifiers can be used.

Right! I missed them

Reply:
Now added